src/app/proposals/LatestProposalsTable.tsx

```diff
@@ -125,10 +127,34 @@ const LatestProposalsTable = ({ proposals }: LatestProposalsTableProps) => {
     }),
   ]
 
-  const [pagination, setPagination] = useState({
-    pageIndex: 0, //initial page index
-    pageSize: 10, //default page size
-  })
+  const router = useRouter()
+  const searchParams = useSearchParams()
+
+  // Convert 1-indexed URL page to 0-indexed internal page
+  const [pagination, setPagination] = useState<PaginationState>(() => ({
+    pageIndex: Math.max(parseInt(searchParams?.get('page') ?? '1') - 1, 0),
+    pageSize: 10,
+  }))
+
+  // Page validation
+  useEffect(() => {
+    const totalPages = Math.ceil(filteredProposalList.length / pagination.pageSize)
+    const requestedPage = parseInt(searchParams?.get('page') ?? '1')
+
+    if (requestedPage > totalPages || requestedPage < 1) {
+      router.push('/404')
```

Comment:
Why not just go to first page?

Reply:
Was a bit conflicted here, but I guess for user consistency, it's probably best to redirect to the first page